### DISCUSSION TITLE
[Feature Request] Link documents via the checkbox

### DISCUSSION THREAD

**Discussion by ritschi86** [paperless-ngx/paperless-ngx#8544] [Feature Requests]
Linking documents is one of the best features for me and I really enjoy using it. However, I find the option of linking by searching for the title a bit cumbersome. Perhaps it would be easy to link documents using the checkboxes. In my example, for example, a document from a law firm and a local court, which logically belong together but are two separate documents.
![image](https://github.com/user-attachments/assets/3d2c713c-19b7-4a2f-8f78-8cfa8b77471f)

**Comment by shamoon** [comment]
It’s an interesting idea I will have a think about how it might work. The screenshot above is one idea but what it’s missing is which custom field is involved (that’s the only way to “link” docs)

**Comment by shamoon** [comment]
Now that I think about it, our next version will contain the ability to bulk assign custom field values (#8428) so I think that will largely meet this need (and I can’t really think of another good way to do this without creating a whole bespoke pathway like your screenshot)

**Reply by ritschi86** [reply]
Hey, I think we're making a mistake. I would find it excellent if I could link two or more documents with each other via the GUI and not have to take the cumbersome route via the title search. Editing the fields in bulk doesn't get me anywhere

**Reply by shamoon** [reply]
I assumed by “link documents” you meant using a “document link” type custom field. Are you suggesting something completely different than that?

**Reply by luisr-ca** [reply]
I think editing via bulk option might work if you could copy a document link url and paste it into the "document link" type custom field, but that doesn't appear to be possible.  (I linked one document to another, then copied the url from the DocLink field and tried pasting it into another document link .. did not work).  

Another option: when filling the field in the bulk edit drop down, if the field is type "document link" then allow the search for other document to happen there.  That could allow reuse of the same search code that's already activated when a user is looking at a specific document.

**Reply by luisr-ca** [reply]
@ritschi86 For clarification .. if you're suggesting that linking be done via the UI and the user clicks on many documents, then are you suggesting that the document link would have a link to EVERY other document except itself?    

Based on your screenshot, I assumed that you would want Document 1 to be linked to Document 2, and vice-versa. And I also assumed that those links _would indeed_ be stored in existing "document link" field type.   That seems straightforward with 2 documents.  But as you add more documents, that approach would require code to look at each document, and place the link in every other document that had been clicked.   Is that what you intended to suggest?

**Reply by ritschi86** [reply]
I hope I can clearly express what I mean by that. The actual extra of the document link should not be changed in terms of logic. Based on my example, it would be a great innovation for me personally if I could link the documents 1:1, as the logic already provides for. The link should then not be created by searching for the document title in the custom field, but automatically in the custom field by selecting the two documents via the selection box  
![image](https://github.com/user-attachments/assets/711d37d4-400a-43b6-90f9-72943ca90acb)

MERRY XMAS @ EVERYONE

**Comment by HiranChaudhuri** [comment]
How many documents would you expect to link to each other? Just two? What happens if you need more than that?

Somehow I believe such links should be possible similar as Jira manages links between issues. You can add any amount of such links and even classify them as "blocks, is blocked by, depends, replaces, ...".

**Comment by tobiasgraeber** [comment]
Could this also be used to "Link" a .eml to the Attachments of that .eml and vice versa @shamoon ?

Can this one here be unlocked, please?:
https://github.com/paperless-ngx/paperless-ngx/discussions/2915

Thanks

**Reply by Finkman** [reply]
It also would benefit a feature request by myself: #8442

**Comment by Tailtinn** [comment]
I would love to see this feature too.
Maybe you could extend the current bulk-edit mode (custom fields -> set values), by offering to add the currently selected documents?

Like this:
![image](https://github.com/user-attachments/assets/99a83a83-2a0d-45a2-8618-c6cffdf0582f)

**Comment by Kabulla** [comment]
Strongly support this CR!
This would satisfy the need to link longer correspondence, make the links visible to the viewer of any of the linked documents and serve for a more stringent overview.

**Comment by andreaswaefler** [comment]
Hi @ritschi86  @shamoon 

Every month we receive the main credit card invoice. In addition to this, there are usually around 20–30 individual invoices that need to be linked to the master invoice.

Each of these invoices needs to be reviewed by a different person, and we also assign cost centers to them afterward. I believe setting up a proper linking process between the individual invoices and the master invoice could be a good solution.

Do you have any suggestions or best practices for handling this more efficiently?

Thanks in advance!

**Comment by mfbloom** [comment]
I was just about to open a new discussion for a feature like this. My use case is associating health claims and benefit information together to see the claims submitted, approval/denials, and the final explanation of benefits. 

This could be super powerful if the classifier could automatically link documents together based on similar/same reference id's found on the document on ingest, etc. Many-to-many document associations is a must.

Would love to see this added!

**Comment by ritschi86** [comment]
Push this Feature 🚀

**Comment by Karthogan** [comment]
With the mass edit of custom fields we can now link a bunch of selected documents to one document with our choice of document link custom field. 

But this will only put that single link value in those sub-documents, with nothing in the main document.

I think this goes back to document links being directional to specific documents, instead of hidden tags/groups or mutual relations. When you want the full set of related documents to clearly be related, it's not ideal that all except one are related to a main document, and there is no list of related documents at all on that main one.

We could have have a "link every selected document to every selected document" function from selection, that makes all the one sided links that suck to manually write out, but that seems more like a troublesome workaround. It's also easy to break, as the link pointing in one direction can be accidentally removed, while its pair pointing back remains. 

If the relations were mutual, then seeing all related documents from any related document would be the default, and maybe it'd also be easier to filter (if we had a "linked to xy document filter, that should bring up the whole bunch). If it was groups instead of mutual links, it'd basically achieve the same result to the user. 

Maybe mutual relations for document linking would't always make the best, but I think having that option would make a lot of sense and improve usability. Including making quickly linking documents together from the UI practical.

Currently the only way to build mutual relations between grouped documents, and filter for the bunch when needed, is still just creating an insane amount of fully visible tags.
